### ISSUE TITLE
Help with another implementation

### ISSUE THREAD

**Issue by josianerodrigues** [flyingpot/pytorch_deephash#4]
Hi @flyingpot 

There is an extension of this work available here: https://github.com/kevinlin311tw/Caffe-DeepBinaryCode. I would like to port to Pytorch, but I do not understand much of Caffe. Could you help me? Where I could start? As it is an extension of this work that you implemented I think I can utilize your implementation. I just need to implement for CIFAR-10, so I guess I just need to implement the final part of the network that generates the hashing codes.

**Comment by spoonnotfound** [comment]
The extension of Kevin's work is adding two loss functions in the network. Since the work of CVPRW15 is great enough, the extension cannot improve the performance a lot. I've tried it with Pytorch, and the performance between the two works is nearly the same. So I don't think I need to add it in this project. If you really need to implement it, I can provide you the codes here.

**Comment by josianerodrigues** [comment]
I really need to implement this work, if you can provide I thank you.

**Comment by josianerodrigues** [comment]
Hi @flyingpot,
Sorry for taking your time, but could you provide the code this week yet? Thank you.

**Comment by spoonnotfound** [comment]
Sorry. I'm a little busy these days. The code is as follows:
```python
k1loss = nn.MSELoss().cuda()
k2loss = nn.MSELoss().cuda()
...
loss = softmaxloss(outputs, targets) - 0.5 * k1loss(feat, Variable((torch.ones(feat.size())*0.5).cuda())) + 0.5 * k2loss(feat.mean(1), Variable((torch.ones(feat.mean(1).size())*0.5).cuda()))
```

**Comment by josianerodrigues** [comment]
Really thank you.
I have just one question. The feat variable is the feature returned by the network?

**Comment by spoonnotfound** [comment]
Yes, it is.
